cai_causal_graph/time_series_causal_graph.py

```diff
@@ -0,0 +1,755 @@
+    @wraps(func)
+    def wrapper(self: TimeSeriesCausalGraph, *args, **kwargs):
+        function = func(self, *args, **kwargs)
+        self._summary_graph = None
+        self._order = None
+        self._variables = None
+        return function
+
+    return wrapper
+
+
+class TimeSeriesCausalGraph(CausalGraph):
+    """
+    A causal graph for time series data.
+
+    The node in a time series causal graph will have additional metadata that
+    gives the time information of the node together with the variable name.
+    The two additional metadata are:
+    - 'timedelta': the time difference with respect the reference time 0
+    - 'variable_name': the name of the variable (without the lag information).
+    """
+
+    def __init__(
+        self,
+        input_list: Optional[List[NodeLike]] = None,
+        output_list: Optional[List[NodeLike]] = None,
+        fully_connected: bool = True,
+        store_graphs: bool = False,
+    ):
+        """
+        Initialize the time series causal graph.
+
+        :param input_list: list of input nodes. Default is None.
+        :param output_list: list of output nodes. Default is None.
+        :param fully_connected: if True, the graph will be fully connected.
+            Default is True.
+        :param store_graphs: if True, the graphs will be stored.
+
+        Example usage:
+            >>> from cai_causal_graph import CausalGraph, TimeSeriesCausalGraph
+            >>>
+            >>>
+            >>> # How to initialize a TimeSeriesCausalGraph from a CausalGraph
+            >>> cg = CausalGraph()
+            >>> cg.add_edge('X1 lag(n=1)', 'X1', edge_type=EDGE_T.DIRECTED_EDGE)
+            >>> cg.add_edge('X2 lag(n=1)', 'X2', edge_type=EDGE_T.DIRECTED_EDGE)
+            >>> cg.add_edge('X1', 'X3', edge_type=EDGE_T.DIRECTED_EDGE)
+            >>>
+            >>> # The time seroes causal graph will have the same nodes and edges as the causal graph
+            >>> # but it is aware of the time information so 'X1 lag(n=1)' and 'X1' represent the same
+            >>> # variable but at different times.
+            >>> ts_cg = TimeSeriesCausalGraph.from_causal_graph(cg)
+
+            >>> # How to initialize a TimeSeriesCausalGraph directly
+            >>> ts_cg = TimeSeriesCausalGraph()
+            >>> ts_cg.add_edge('X1 lag(n=1)', 'X1', edge_type=EDGE_T.DIRECTED_EDGE)
+            >>> ts_cg.add_edge('X2 lag(n=1)', 'X2', edge_type=EDGE_T.DIRECTED_EDGE)
+            >>> ts_cg.add_edge('X1', 'X3', edge_type=EDGE_T.DIRECTED_EDGE)
+        """
+        super().__init__(input_list, output_list, fully_connected)
+
+        self._store_graphs = store_graphs
+        # autoregressive order of the graph (max lag)
+        self._order: Optional[int] = None
+        # list of variables in the graph, i.e. discarding the lags (X1(t-1) and X1 are the same variable)
+        self._variables: Optional[List[str]] = None
```

Comment:
I don't think we necessarily have one, as you can instantiate and build up `TimeSeriesCausalGraph` by itself without a `CausalGraph` instance, so I think this is fine as it is

Reply:
Even if its an empty graph you will still have a graph. Not a big deal but just feels cleaner to instantiate here, especially in case where we pass in features and do fully connected for example on instantiation